feat(hooks): Error hooks (#7349)
videojs/video.js: docs/guides/hooks.md

videojs('some-id', {autoplay: true, controls: true});
```

### beforeerror

`beforeerror` occurs just as we get an error on the player. This allows plugins or other custom code to intercept the error and modify it to be something else.

> `error` is can be an object such as `{code: 4}` or something with a message. It potentially includes a `cause` property too. Alternatively, it could be `null` which means that the current error will be getting cleared.

Reviewer: ```suggestion
`error` can be a string with a message, an object or `MediaError`, a javascript `Error`, a string, or null indicating that error should be cleared.

Examples:
MediaError or Object:
`{code: 4, message: 'something', cause: 'xyz'}`. 

String
`something bad happened`

No error:
`null`

> `error` is can be an object such as `{code: 4}` or something with a message. It potentially includes a `cause` property too. Alternatively, it could be `null` which means that the current error will be getting cleared.

Reviewer: ```suggestion
`error` is an object that has a `code` and `message` property and potentially a `cause` property too. Alternatively, it could be `null` which means that the current error should be cleared.
```

Author: Looks like error can be a string or just a number too, so, I'll update this and the code based on expectations.
